Packs/ThreatExchange/Integrations/ThreatExchangeV2/ThreatExchangeV2.py

```diff
@@ -0,0 +1,917 @@
+"""
+An integration module for the ThreatExchange V2 API.
+API Documentation:
+    https://developers.facebook.com/docs/threat-exchange/reference/apis
+"""
+
+import collections
+from typing import Tuple
+import urllib3
+from CommonServerUserPython import *  # noqa
+from CommonServerPython import *  # noqa # pylint: disable=unused-wildcard-import
+
+# Disable insecure warnings #
+urllib3.disable_warnings()  # pylint: disable=no-member
+
+DEFAULT_LIMIT = 20
+COMMAND_PREFIX = 'threatexchange'
+VENDOR_NAME = 'ThreatExchange v2'
+CONTEXT_PREFIX = 'ThreatExchange'
+THREAT_DESCRIPTORS_SUFFIX = 'threat_descriptors'
+MALWARE_ANALYSES_SUFFIX = 'malware_analyses'
+THREAT_TAGS_SUFFIX = 'threat_tags'
+TAGGED_OBJECTS_SUFFIX = 'tagged_objects'
+THREAT_EXCHANGE_MEMBERS_SUFFIX = 'threat_exchange_members'
+TIMEOUT_FOR_LIST_CALLS = 30
+
+
+class ThreatExchangeV2Status:
+    UNKNOWN = 'UNKNOWN'
+    NON_MALICIOUS = 'NON_MALICIOUS'
+    SUSPICIOUS = 'SUSPICIOUS'
+    MALICIOUS = 'MALICIOUS'
+
+
+class Client(BaseClient):
+    """Client class to interact with the service API
+
+    This Client implements API calls, and does not contain any XSOAR logic.
+    Should only do requests and return data.
+    It inherits from BaseClient defined in CommonServer Python.
+    Most calls use _http_request() that handles proxy, SSL verification, etc.
+    For this implementation, no special attributes defined
+    """
+
+    def __init__(self, base_url, access_token, verify=True, proxy=False):
+        super().__init__(base_url, verify, proxy)
+        self.access_token = access_token
+
+    def ip(self, ip: str) -> Dict:
+        """
+        See Also:
+            https://developers.facebook.com/docs/threat-exchange/reference/apis/threat-descriptors
+        Args:
+            ip: ip address
+
+        Returns: The API call response
+        """
+        response = self._http_request(
+            'GET',
+            THREAT_DESCRIPTORS_SUFFIX,
+            params={
+                'access_token': self.access_token,
+                'type': 'IP_ADDRESS',
+                'text': ip,
+                'strict_text': True
```

Comment:
shouldnt `strict_text` be configurable ?

Reply:
In V1 'strict_text' was set to true in all reputation commands. 
I think it make sense because usually when a user wants to run a reputation command for a certain indicator, he would want to get results for this specific indicator and not for approximate matching indicators  - therefore 'strict_text' field should be set to true by default and shouldn't be configurable. 

Only for query command this field is configurable for the user, and set to false by default.